server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -417,6 +508,7 @@ public KeywordFieldType(
             this.scriptValues = builder.scriptValues();
             this.isDimension = builder.dimension.getValue();
             this.isSyntheticSource = isSyntheticSource;
+            this.hasDocValuesSparseIndex = DocValuesSkipIndexType.NONE.equals(fieldType.docValuesSkipIndexType()) == false;
```

Comment:
Checking this way we make sure the boolean value is correct even if in future new skip indices other than `RANGE` are introduced in Lucene.